### PR TITLE
Tests and error handling

### PR DESCRIPTION
**Pull request by jstavh**
Tests and error handling! So far within the ledger directory. Refactor the mod.rs slightly with error handling. Also fixes the clippy warning.

### CODE REVIEW

**Review by Isaac-DeFrain — COMMENTED** [review]
LGTM!
Can you include a test comparing the `AccountDiff` from a block against the post balance calculation? I suggest using the genesis ledger and block 2 (as long as it has payments in it). We don't need to check all balances, just one that interacts with a payment should do

**Review by jenr24G — CHANGES_REQUESTED** [review]
Looks great! I agree with Isaac about using the post balances as a source of truth! Can you fix the clippy warning? Thanks!

**Comment by jstavh** [comment]
@Isaac-DeFrain @jenr24G To keep our PR's small, let's merge this one since you both think it looks good and then make a separate PR for the test you mention and also one for the `from_block_fees` function?

File: src/state/ledger/diff/account.rs [hunk 1]
```diff
@@ -162,3 +162,132 @@ impl std::fmt::Debug for UpdateType {
...
+
+        let account_diff =
+            AccountDiff::from_coinbase(coinbase_receiver_result.into(), supercharge_coinbase);
+
+        let expected_payment_diff = PaymentDiff {
+            public_key: coinbase_receiver.into(),
+            amount: 1440 * (1e9 as u64),
+            update_type: UpdateType::Deposit,
+        };
+        let expected_account_diff = AccountDiff::Payment(expected_payment_diff);
```
Comment (Isaac-DeFrain) [inline src/state/ledger/diff/account.rs:247]: Interesting. This could explain why our nonces were off.

File: src/state/ledger/mod.rs [hunk 2]
```diff
@@ -92,37 +98,48 @@ impl Ledger {
...
+                            Account::from_delegation(
+                                account_before,
+                                delegation_diff.delegate.clone(),
+                            )
+                        }
+                    };
+
+                    self.accounts.insert(diff.public_key(), account_after);
+                }
+                None => {
```
Comment (Isaac-DeFrain) [inline src/state/ledger/mod.rs:132]: How does this work with the creation of a new account?

File: src/state/ledger/diff/account.rs (hunk 1 above)
Comment (Isaac-DeFrain) [reply src/state/ledger/diff/account.rs:247]: We're processing coinbases as payments? Payments should increase the nonce, but I don't think coinbases should. Not sure. Just a thought.

**Review by Isaac-DeFrain — APPROVED** [review]

**Review by jenr24G — APPROVED** [review]

**Pull request merged**